Applied-Upstream: commit: d1db64291475fe9b83b2fca75c793b33859ac3c9
Author: Evan Maddock <5157277+EbonJaeger@users.noreply.github.com>
Last-Update: 2022-03-08
Description: [PATCH] Fix inconsistency when opening Raven (#68)
 Per the answer here (https://stackoverflow.com/questions/22178524/gtk-named-stack-childs),
 if a stack child is not visible, asking the stack to set the visible child will silently
 fail, apparently even if the child being shown *is* visible.
 The NotificationView widget already calls show_all() in its constructor, so the call only
 had to be added to the applet window. This fixes the case where clicking the notification
 bell fails to open the correct page, and possibly also fixes where clicking the Raven
 trigger for the first time sometimes fails to open Raven (it is technically open, but
 the elements aren't visible).
 Ref #31
---
 src/raven/main_view.vala | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/raven/main_view.vala b/src/raven/main_view.vala
index 23628de9..b0e55003 100644
--- a/src/raven/main_view.vala
+++ b/src/raven/main_view.vala
@@ -77,6 +77,10 @@ namespace Budgie {
 			mpris = new MprisWidget();
 			box.pack_start(mpris, false, false, 0);
 
+			// Make sure everything is shown. Not having this can cause
+			// silent failures when switching stack pages or opening Raven.
+			scroll.show_all();
+
 			main_stack.notify["visible-child-name"].connect(on_name_change);
 			set_clean();
 			set_sound_widget_events();
-- 
2.34.1

